diff --git a/src/Helpers_TEST.cc b/src/Helpers_TEST.cc
index a602f210..386d8188 100644
--- a/src/Helpers_TEST.cc
+++ b/src/Helpers_TEST.cc
@@ -1008,6 +1008,8 @@ TEST(HelpersTest, AppendToStream)
   math::appendToStream(out, pi);
 #if defined(_WIN32) && !defined(__amd64__) && !defined(__i386__)
   EXPECT_EQ(out.str(), "0 456 0 3.14159 3.141592654 3.141592653589793");
+#elif  __aarch64__
+  EXPECT_EQ(out.str(), "0 456 0 3.14159 3.141592654 3.141592653589793238462643383279503");
 #else
   EXPECT_EQ(out.str(), "0 456 0 3.14159 3.141592654 3.141592653589793239");
 #endif
@@ -1018,6 +1020,8 @@ TEST(HelpersTest, AppendToStream)
   math::appendToStream(out, pi);
 #if defined(_WIN32) && !defined(__amd64__) && !defined(__i386__)
   EXPECT_EQ(out.str(), "0 456 0 3.14159 3.141592654 3.141592653589793 3.14");
+#elif  __aarch64__
+  EXPECT_EQ(out.str(), "0 456 0 3.14159 3.141592654 3.141592653589793238462643383279503 3.14");
 #else
   EXPECT_EQ(out.str(), "0 456 0 3.14159 3.141592654 3.141592653589793239 3.14");
 #endif
